### ISSUE TITLE
Reduced shares received by pool with Claymore 9.7 + proxy 1.1

### ISSUE THREAD

**Issue by jimmykl** [JuicyPasta/Claymore-No-Fee-Proxy#20]
I'm running Claymore 9.7 with the new v1.1 proxy, using fake WAN addresses and mining at ethermine.org. I restarted my rigs after setting it all up and the good news is I don't get the local proxy warning any more and the fee address swap is working.

But the number of shares received by the pool has dropped (see attached screenshot). I understand the effective hashrate varies a lot and Claymore and the pool's reported rate are the same but the actual shares received in the bottom graph before using the proxy were ~380–430 and with proxy I'm getting ~330–390.

Is this expected? Maybe something to do with how Claymore processed his fee's shares? Or is there a known slowdown due to using the proxy? Or something else?

**Comment by drdada** [comment]
Hello, 
No obviously, this is not expected. Did you set some DNS redirection ?

**Comment by jimmykl** [comment]
Not that I'm aware of… the rigs are using google dns 8.8.8.8

**Comment by drdada** [comment]
It sounds good.
Are you familiar with Linux ? I would recommand to build a linux gateway and redirect devfee via this way. I also detect a hashrate drop sometimes and i'm wondering if the proxy is still undetected.
I'm still testing this "gateway tricks"

**Comment by drdada** [comment]
Can you send me a PM on gitter please, we will try something together.

**Comment by jimmykl** [comment]
I am familiar with some aspects of linux but not sure how to build a gateway. I have a raspberry pi I use for pihole (dns based ad blocking) and it would be perfect for this as it's on all the time. I'll pm you shortly

**Comment by paullee29** [comment]
Hi DrDada
Thanks for your work, but I think I am having the same problem, it seems that when I am redirecting the packets, I get a reduced share count and a reduced effective hashrate, even though it reports the proper one to ethermine.

I will test again and will do 24 hour testing on each and report back.

**Comment by jimmykl** [comment]
Yes I tried a virtual network adapter and redirecting the port and both seem to get picked up. It looks like Claymore has a foolproof way to detect when this is happening and reduce the shares. I'll probably just pay the fee when dual mining or use ethminer.

**Comment by khaman1** [comment]
Try Claymore 9.8. They also claim that their newest code willl do much less invalid and outdated shares

**Comment by drdada** [comment]
I'm aware of the detecting issue.
We are working on it. 
ATM you have choices:
- Pay the fee
- Use the -nofee 1 arg (and still having reduce share)
- Use [Ethminer ](https://github.com/ethereum-mining/ethminer/releases)(free but with lower hashrate speed) - You can also use [MinerLamp](https://github.com/orkblutt/MinerLamp) as addon.

**Comment by khaman1** [comment]
I have been using Ethminer in a month and I'm pretty sure 99% people will say ethminer hashrate is always higher than or probably equal to Claymore. The problem of ethminer is no watchdog but we can run a code to check the GPU states continuously. I also have the share stale when using your code but I didn't check the loss. I'm switching back to ethminer. Still ethminer is an open source and contributed by community, so it would be intuitively better.

**Comment by khaman1** [comment]
I didn't try the dual mining of Claymore. Its latest version claims no effect on Eth hashrate when mining another coin during the "extra cycle" (I don't get this term btw). But it seems like ethminer uses that to mine more Eth because the reported hashrate after submitting the share is remarkably higher.

**Comment by drdada** [comment]
@khaman1 Try MinerLamp (see link below) if you want watchdog for Ethminer.

**Comment by khaman1** [comment]
I have written a short [code](https://github.com/khaman1/Ethminer-Watchdog/blob/master/resetifnotenoughgpu.py) to monitor it. My rigs sometimes get GPU error and have to restart the computer, not just the application. Not pretty sure how MinerLamp handles that but doing restart and copy the shortcut of the code and ethminer to Start-up application (Run shell:startup) can do the job.
Anyway, I will try MinerLamp

**Comment by MichaelA2014** [comment]
Is there any benefits of 9.8 over 9.6? I am using 9.6 and it's churning away just fine. No noticable drop. Of course maybe I was loosing shares all this time on Nanopool? Who knows...

**Comment by vctender** [comment]
Any updates? How is the proxy (technical) recognized by claymore?

**Comment by sb3rg** [comment]
Seeing warning from Claymore miner v11.1 that using a stratum proxy will result in more stale shares.  I'll let it run over night to see if it's legit, but perhaps someone else on this feed has already tested it?  And update would be very helpful. Thank you!

**Comment by drdada** [comment]
@sb3rg If you got warning it means Claymore spotted the proxy and apply some slowdown (stale share, reduced shares or higher power consumption). https://github.com/JuicyPasta/Claymore-No-Fee-Proxy#claymore-warns-me-something-about-local-proxy
If you did the Wan trick you should bypass the detection system (and not having warnings).
So v11.1 (and probably higher versions) are not compatible anymore.

@MichaelA2014 Try to mine with ethermine, you'll got more stats about stales shares.

@vctender Since the software is closed-source we don't know how it detect proxies.
The only way to know that is via multiples tests (wireshark, process monitor, ...).

**Comment by JsBergbau** [comment]
This are my rates with devfee removal Proxy. I've used another so far, however it is not compatible with v11.2 so I've now switched to your python proxy. Switch occured right now, so lets see. But I think Claymore can't detect it like so far right now. The whole occured when switchting to 11.2, Did some other maintanance stuff at well-
However I have sometimes the same low effective hashrates as  jimmykl after running proxy. I think its another issue, because when claymore would detect my proxy, it would always give lower hash rates and not only sometimes.

![rates030318](https://user-images.githubusercontent.com/37013344/36932435-14163bf0-1ec9-11e8-90d0-135da5dca4de.jpg)
